<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PortConfigEditDialog</class>
 <widget class="QDialog" name="PortConfigEditDialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>600</width>
    <height>550</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>端口配置</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="groupBasicInfo">
     <property name="title">
      <string>基本信息</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <widget class="QLabel" name="labelFuncBlock">
        <property name="text">
         <string>功能子块:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QLabel" name="valueFuncBlock">
        <property name="text">
         <string/>
        </property>
        <property name="frameShape">
         <enum>QFrame::Panel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelPortName">
        <property name="text">
         <string>端口号:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QLabel" name="valuePortName">
        <property name="text">
         <string/>
        </property>
        <property name="frameShape">
         <enum>QFrame::Panel</enum>
        </property>
        <property name="frameShadow">
         <enum>QFrame::Sunken</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupPortConfig">
     <property name="title">
      <string>端口配置</string>
     </property>
     <layout class="QFormLayout" name="formLayout_2">
      <item row="0" column="0">
       <widget class="QLabel" name="labelPortType">
        <property name="text">
         <string>端口类型:</string>
        </property>
       </widget>
      </item>
      <item row="0" column="1">
       <widget class="QComboBox" name="comboPortType">
        <item>
         <property name="text">
          <string>electric</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>hydraulic</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>mechanical</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>other</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="1" column="0">
       <widget class="QLabel" name="labelDirection">
        <property name="text">
         <string>方向:</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QComboBox" name="comboDirection">
        <item>
         <property name="text">
          <string>input</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>output</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>bidirectional</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>internal</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="2" column="0">
       <widget class="QLabel" name="labelVariableProfile">
        <property name="text">
         <string>变量配置:</string>
        </property>
       </widget>
      </item>
      <item row="2" column="1">
       <widget class="QComboBox" name="comboVariableProfile">
        <item>
         <property name="text">
          <string>default</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>custom</string>
         </property>
        </item>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="labelVariables">
        <property name="text">
         <string>变量列表:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QLineEdit" name="editVariables">
        <property name="placeholderText">
         <string>用逗号分隔，如: i,u</string>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="labelConnectMacro">
        <property name="text">
         <string>连接宏:</string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QComboBox" name="comboConnectMacro">
        <property name="editable">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupMacroList">
     <property name="title">
      <string>可用连接宏族</string>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_2">
      <item>
       <widget class="QTableWidget" name="tableMacros">
        <property name="editTriggers">
         <set>QAbstractItemView::NoEditTriggers</set>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::SingleSelection</enum>
        </property>
        <column>
         <property name="text">
          <string>宏族名称</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>领域</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>支持端口数</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>描述</string>
         </property>
        </column>
       </widget>
      </item>
      <item>
       <layout class="QHBoxLayout" name="horizontalLayoutMacroButtons">
        <item>
         <widget class="QPushButton" name="btnAddMacroFamily">
          <property name="text">
           <string>添加宏族</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QPushButton" name="btnDeleteMacroFamily">
          <property name="text">
           <string>删除宏族</string>
          </property>
         </widget>
        </item>
        <item>
         <spacer name="horizontalSpacerMacro">
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
          <property name="sizeHint" stdset="0">
           <size>
            <width>40</width>
            <height>20</height>
           </size>
          </property>
         </spacer>
        </item>
       </layout>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <spacer name="horizontalSpacer">
       <property name="orientation">
        <enum>Qt::Horizontal</enum>
       </property>
       <property name="sizeHint" stdset="0">
        <size>
         <width>40</width>
         <height>20</height>
        </size>
       </property>
      </spacer>
     </item>
     <item>
      <widget class="QPushButton" name="btnOk">
       <property name="text">
        <string>确定</string>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="btnCancel">
       <property name="text">
        <string>取消</string>
       </property>
      </widget>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>btnOk</sender>
   <signal>clicked()</signal>
   <receiver>PortConfigEditDialog</receiver>
   <slot>accept()</slot>
  </connection>
  <connection>
   <sender>btnCancel</sender>
   <signal>clicked()</signal>
   <receiver>PortConfigEditDialog</receiver>
   <slot>reject()</slot>
  </connection>
 </connections>
</ui>
